usr/sbin/p11sak/p11sak.c

```diff
@@ -912,17 +913,17 @@ static CK_ULONG char2attrtype(char c)
     }
 }
 
-static void set_bool_attr_from_string(CK_ATTRIBUTE *attr, char *attr_string)
+static void set_bool_attr_from_string(CK_ATTRIBUTE *attr, char attr_string)
 {
-    int i;
-
-    if (!attr_string)
+    if (!attr_string) {
         return;
-
-    for (i = 0; i < (int) strlen(attr_string); i++) {
-        if (char2attrtype(attr_string[i]) == attr->type) {
-            attr->pValue = &ckb_true;
-        }
+    }
+    attr->type = char2attrtype(toupper(attr_string));
+    attr->ulValueLen = sizeof(CK_BBOOL);
+    if (isupper(attr_string) == 0) {
+        attr->pValue = &ckb_false;
+    } else {
+        attr->pValue = &ckb_true;
```

Comment:
I like this, but the upper/lower case specification is currently only for symmetric keys, right? I would then like to have the same behavior for the the asymmetric keys as well. 

Maybe you can even use one single function for both cases, symmetric and asymmetric keys?  For asymmetric keys ist function `set_battr()` currently, while for symmetric keys ist function `set_bool_attr_from_string()`. I guess it would make sense to unify both key types as much as possible, since its essentially the same task that is performed.

Reply:
Done.